src/main/java/com/authservice/authservice/controller/AuthenticationController.java

```diff
@@ -49,6 +52,18 @@ public AuthenticationController(AuthenticationManager authenticationManager, Jwt
         return ResponseEntity.ok(new JWTResponse(token));
     }
 
+    /**
+     * This function will validate the token and return the userId
+     *
+     * @return
+     */
+    @GetMapping("/validating_token")
+    public ResponseEntity<AuthenticatedResponse> validatingToken() {
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        User currentUser = (User) auth.getPrincipal();
+        return ResponseEntity.ok().body(new AuthenticatedResponse(currentUser.getId()));
```

Comment:
use builder pattern here instead.

Reply:
I can't apply @Builder on the top of the class as Spring security User class has it's own builder (Incompatible builder types in the child class and parent class), and i can't apply builder on the constructor also as i don't have all the user info like password and all. So i have created a static method and applied @Builder on the top of that method which will return loggedIn user Id.